perfmetrics/scripts/hns_rename_folders_metrics/generate_folders_and_files.py

```diff
@@ -95,7 +95,67 @@ def list_directory(path) -> list:
     return contents_url
   except subprocess.CalledProcessError as e:
     logmessage(e.output.decode('utf-8'))
-    subprocess.call('bash', shell=True)
+
+
+def check_if_dir_structure_exists(directory_structure) -> (int):
+  """Checks if the directory structure mentioned in the config file already
+  exists in the GCS bucket.
+
+  Args:
+    directory_structure: Json Object representing the directory structure.
+
+  Returns:
+    true if the existing structure in GCS bucket exactly matches with teh config
+     file
+    false otherwise
+  """
+  bucket_name = directory_structure["name"]
+  bucket_url = 'gs://{}'.format(bucket_name)
+
+  # check for top level folders
+  folders = list_directory(bucket_url)
+  nested_folder_count = "nested_folders" in directory_structure
+  if "folders" in directory_structure:
+    if len(folders) != directory_structure["folders"][
+      "num_folders"] + nested_folder_count:
+      return 0
+
+    # for each non-nested folder , check the count of files
+    for folder in directory_structure["folders"]["folder_structure"]:
+      try:
+        files = list_directory('{}/{}'.format(bucket_url, folder["name"]))
+        if len(files) != folder["num_files"]:
+          return 0
+      except:
+          # if the list directory fails wth url did not match object, folder
+          # specified in config file does not exist in bucket.
+          return 0
+
+  # check the number of second level folders in nested folders
+  if nested_folder_count:
+    nested_folder = directory_structure["nested_folders"]["folder_name"]
+    try:
+      second_level_folders = list_directory(
+          '{}/{}'.format(bucket_url, nested_folder))
+      if len(second_level_folders) != directory_structure["nested_folders"][
+        "num_folders"]:
+        return 0
+
+      # for each second level folder in "nested" folders, check the count of files
+      for folder in directory_structure["nested_folders"]["folder_structure"]:
```

Comment:
If possible please create common function to verify directory structure.

Reply:
Done.